[ARM/Linux] Build Failure on ARM

Recent commit (9c2f1fe) breaks ARM build.
The (len) variable need explicit type conversion to unsigned value.

```
/home/jyoung/git/dotnet/corefx/src/Native/System.Native/pal_networkchange.cpp:63:63: error: comparison of integers of
      different signs: '__u32' (aka 'unsigned int') and 'ssize_t' (aka 'int') [-Werror,-Wsign-compare]
    for (nlmsghdr* hdr = reinterpret_cast<nlmsghdr*>(buffer); NLMSG_OK(hdr, len); NLMSG_NEXT(hdr, len))
                                                              ^~~~~~~~~~~~~~~~~~
/home/jyoung/git/dotnet/rootfs-corefx/arm/usr/include/linux/netlink.h:89:24: note: expanded from macro 'NLMSG_OK'
                           (nlh)->nlmsg_len <= (len))
                           ~~~~~~~~~~~~~~~~ ^
1 error generated.
```

```
 55     ssize_t len; 
... SKIP ...
 63     for (nlmsghdr* hdr = reinterpret_cast<nlmsghdr*>(buffer); NLMSG_OK(hdr, len); NLMSG_NEXT(hdr, len))
```

```
 87 #define NLMSG_OK(nlh,len) ((len) >= (int)sizeof(struct nlmsghdr) && \           
 88                (nlh)->nlmsg_len >= sizeof(struct nlmsghdr) && \                 
 89                (nlh)->nlmsg_len <= (len))   
```

I will post a PR soon.

--- a/src/Native/System.Native/pal_networkchange.cpp
+++ b/src/Native/System.Native/pal_networkchange.cpp
@@ -60,7 +60,7 @@ extern "C" void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetwo
         return;
     }
 
-    for (nlmsghdr* hdr = reinterpret_cast<nlmsghdr*>(buffer); NLMSG_OK(hdr, len); NLMSG_NEXT(hdr, len))
+    for (nlmsghdr* hdr = reinterpret_cast<nlmsghdr*>(buffer); NLMSG_OK(hdr, UnsignedCast(len)); NLMSG_NEXT(hdr, len))
     {
         switch (hdr->nlmsg_type)
         {

